addons/mail/static/src/core/common/thread_model.js

```diff
@@ -708,6 +708,11 @@ export class Thread extends Record {
         return cw;
     }
 
+    closeChatWindow({ noLeaveChannel = false, force = false } = {}) {
+        const chatWindow = this.store.ChatWindow.get({ thread: this });
+        chatWindow?.close({ notifyState: false, noLeaveChannel, force });
+    }
+
```

Comment:
I feel that `notifyState: false` and `noLeaveChannel` should work together.

I thin you can avoid introducing `noleaveChannel` and instead rely on `notifyState` for the leaveChannel override of `_onClose`.